src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.Dummy.cs

```diff
@@ -50,6 +52,9 @@ public override CSharpParseOptions Options
                 get { return CSharpParseOptions.Default; }
             }
 
+            public override ImmutableDictionary<string, ReportDiagnostic> DiagnosticOptions
+                => throw ExceptionUtilities.Unreachable;
```

Comment:
>               => throw ExceptionUtilities.Unreachable; [](start = 1, length = 55)

Why unreachable over invalidoperation?

Reply:
InvalidOperation is sometimes thrown by framework assemblies. Unreachable is only thrown by us and always indicates a logic bug, so I prefer to use it for methods that we should never be calling.